web/src/stream_data.js

```diff
@@ -351,24 +351,27 @@ export function muted_stream_ids() {
         .map((sub) => sub.stream_id);
 }
 
-export function get_subscribed_streams_for_user(user_id) {
+export function get_sub_or_unsub_streams_for_user(user_id, get_unsubscribed = false) {
     // Note that we only have access to subscribers of some streams
     // depending on our role.
     const all_subs = get_unsorted_subs();
-    const subscribed_subs = [];
+    const streams = [];
     for (const sub of all_subs) {
         if (!can_view_subscribers(sub)) {
             // Private streams that we have been removed from appear
             // in get_unsorted_subs; we don't attempt to check their
             // subscribers (which would trigger a warning).
             continue;
         }
-        if (is_user_subscribed(sub.stream_id, user_id)) {
-            subscribed_subs.push(sub);
+        if (
+            (is_user_subscribed(sub.stream_id, user_id) && !get_unsubscribed) ||
+            (!is_user_subscribed(sub.stream_id, user_id) && get_unsubscribed)
+        ) {
+            streams.push(sub);
         }
     }
 
-    return subscribed_subs;
+    return streams;
```

Comment:
While this may reduce code duplication, I would prefer we had separate function for `get_subscribed_streams_for_user` and `get_unsubscribed_streams_for_user` for reading claring in code.